ts/features/bonus/cgn/components/merchants/CgnMerchantsDiscountItem.tsx

```diff
@@ -57,19 +57,51 @@ const CgnMerchantDiscountItem: React.FunctionComponent<Props> = ({
               </H4>
             </View>
             <View spacer xsmall />
-            {lookup(0, [...discount.productCategories]).fold(
-              undefined,
-              categoryKey =>
-                getCategorySpecs(categoryKey).fold(undefined, c => (
-                  <View style={styles.row}>
-                    {c.icon({ height: 22, width: 22, fill: IOColors.bluegrey })}
-                    <View hspacer small />
-                    <H5 weight={"SemiBold"} color={"bluegrey"}>
-                      {I18n.t(c.nameKey).toLocaleUpperCase()}
-                    </H5>
-                  </View>
-                ))
-            )}
+            {discount.productCategories.length > 2
+              ? lookup(0, [...discount.productCategories]).fold(
+                  undefined,
+                  c => (
+                    <>
+                      {renderCategoryElement(c)}
+                      <View
+                        hspacer
+                        small
+                        style={{
+                          borderRightColor: IOColors.bluegrey,
+                          borderRightWidth: 1
+                        }}
+                      />
+                      <View hspacer small />
+                      <H5 color={"bluegrey"} weight={"SemiBold"}>
+                        {I18n.t(
+                          "bonus.cgn.merchantDetail.categories.counting",
+                          {
+                            count: discount.productCategories.length - 1
+                          }
+                        ).toLocaleUpperCase()}
+                      </H5>
+                    </>
+                  )
+                )
+              : discount.productCategories.map(category => (
+                  <>
```

Comment:
```suggestion
              : discount.productCategories.map((category, i) => (
                  <View key={i}>
```
Like the previous comment, in order to have unique keys for react children.